lib/utils/stream_extension.dart

```diff
@@ -45,4 +47,44 @@ extension StreamExtension on Stream {
     };
     return controller.stream;
   }
+
+  Stream<SyncUpdate> rateLimitWithSyncUpdate(Duration t) {
```

Comment:
Could you write an ADR or a comment in the code to explain this please ?

Reply:
done